File: freecodecamp/2026-01-09.js
// Helper: Check if a number is prime
function isPrime(num) {
    if (num < 2) return false;
    if (num === 2) return true;
    if (num % 2 === 0) return false;
    
    for (let i = 3; i <= Math.sqrt(num); i += 2) {
        if (num % i === 0) return false;
    }
    return true;
}

// Helper: Get all rotations of a number
function getRotations(num) {
    const str = num.toString();
    const rotations = [];
    
    for (let i = 0; i < str.length; i++) {
        const rotated = str.slice(i) + str.slice(0, i);
        rotations.push(parseInt(rotated));
    }
    
    return rotations;
}

function isCircularPrime(n) {    
    // Check if all rotations are prime
    const rotations = getRotations(n);
    return rotations.every(rotation => isPrime(rotation));
}

File: freecodecamp/2026-01-09.py
def is_prime(num):
    # Check if a number is prime
    if num < 2:
        return False
    if num == 2:
        return True
    if num % 2 == 0:
        return False
    
    i = 3
    while i <= num**0.5:
        if num % i == 0:
            return False
        i += 2
    
    return True


def get_rotations(num):
    # Get all rotations of a number
    str_num = str(num)
    rotations = []
    
    for i in range(len(str_num)):
        rotated = str_num[i:] + str_num[:i]
        rotations.append(int(rotated))
    
    return rotations


def is_circular_prime(n):
    # Check if all rotations are prime
    rotations = get_rotations(n)

    result = False;

    for i in range(len(rotations)):
        result = is_prime(rotations[i])
    
    return result
